apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts

```diff
@@ -405,6 +421,10 @@ export class ParseEventRequest {
   private isBlockedEmail(email: string): boolean {
     return BLOCKED_DOMAINS.some((domain) => email.includes(domain));
   }
+
+  private isGracefulTriggerAllowed(source?: string): boolean {
+    return source === TriggerSourceEnum.DASHBOARD;
```

Comment:
@djabarovgeorge, why do we need to check the source for this? I suggest just using the status for now.

Reply:
please see the description of the PR 

> At the same time, to improve the user experience, we're allowing users to trigger the "Trigger" action directly from the Dashboard. This helps them quickly identify issues while debugging the workflow.

Reply:
Let me know what you think, i am fine removing it because now the trigger is less predicted, but in the same time it will cost workflow debugging flow.

Reply:
I tend to agree with sok, I think trigger should be predictable and throw an error when in a bad step, not the type of things you want to learn about in prod

Reply:
Following on this, @djabarovgeorge is right. Without a more sophisticated way to test an inactive workflow from the Dashboard we will need this hack in the logic to allow Novu users to test workflows from the Dashboard.

But this means we must ensure the API request comes from the Dashboard. That is, we should verify the source and the fact that a JWT (not an API key) was used for authentication.

Reply:
How do we feel about adding additional indicators for this case?  
1. If the status is `ERROR`, we could either:  
   - Change the button color, or  
   - Add a warning icon to highlight the workflow status.  
2. If the status is `ERROR`, the payload could include an extra parameter to show the trigger override, for example:  
```
payload:
{
  "__debugModeBypassStatus": true,
  ...otherParams
}
```